specification/redis/resource-manager/Microsoft.Cache/preview/2023-05-01/redis.json

```diff
@@ -1850,6 +1850,536 @@
           }
         }
       }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Cache/redis/{cacheName}/accessPolicies/{accessPolicyName}": {
+      "put": {
+        "tags": [
+          "Redis",
+          "AAD"
+        ],
+        "operationId": "AccessPolicy_CreateUpdate",
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "azure-async-operation"
+        },
+        "x-ms-examples": {
+          "RedisCacheAccessPolicyCreateUpdate": {
+            "$ref": "./examples/RedisCacheAccessPolicyCreateUpdate.json"
+          }
+        },
+        "description": "Adds an access policy to the redis cache",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "name": "cacheName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the Redis cache.",
+            "pattern": ".*"
+          },
+          {
+            "name": "accessPolicyName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the access policy that is being added to the Redis cache.",
+            "pattern": "^[a-zA-Z0-9]+$"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/RedisCacheAccessPolicyCreateUpdateParameters"
+            },
+            "description": "Parameters supplied to the Create Update Access Policy operation."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
```

Comment:
Prefer to use
```
         {
           "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
         },
         {
           "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
         },
```

Reply:
Refactored specification for using common types api version and subscription id